Containerfile.rag-content

```diff
@@ -0,0 +1,29 @@
+FROM registry.access.redhat.com/ubi9/python-311 as lightspeed-rag-builder
+
+USER 0
+RUN dnf install -y --nodocs --setopt=keepcache=0 --setopt=tsflags=nodocs git ruby gem
+RUN pip install torch --index-url https://download.pytorch.org/whl/cpu
+RUN pip install PyYAML huggingface_hub llama_index langchain_community \
+                faiss-cpu llama-index-vector-stores-faiss sentence_transformers \
+                llama-index-embeddings-huggingface
+RUN gem install asciidoctor
+
+WORKDIR /workdir
+RUN git clone --single-branch --branch enterprise-4.14 https://github.com/openshift/openshift-docs.git
+COPY scripts/asciidoctor-text ./asciidoctor-text
+RUN python asciidoctor-text/convert-it-all.py \
+           -i openshift-docs \
+           -t openshift-docs/_topic_maps/_topic_map.yml \
+           -o ocp-product-docs-plaintext \
+           -a asciidoctor-text/attributes.yaml
+
+COPY scripts/download_embeddings_model.py .
+RUN python download_embeddings_model.py ./embeddings_model
+
+COPY scripts/generate_embeddings.py .
+RUN python generate_embeddings.py -f ocp-product-docs-plaintext -m embeddings_model -o vector_db
```

Comment:
i'd consider introducing an ocp version in the vector_db path.  and maybe a type, too.  So like:

`vector_db/ocp_docs/4.15`

Reply:
does this generate embedding metadata to go along w/ the db?  (info about the source content, the embedding model used, etc?)

Reply:
It does, but embedding-model can be improved:
```
$ cat vector_db/metadata.json | jq
{
  "execution-time": 4036.091420173645,
  "llm": "None",
  "embedding-model": "embeddings_model",
  "index_id": "product",
  "vector-db": "faiss",
  "chunk": 1500,
  "overlap": 10,
  "total-embedded-files": 13996
}

```

Reply:
Corrected